src/libs/actions/PersonalDetails.js

```diff
@@ -236,6 +237,53 @@ function setPersonalDetails(details) {
     mergeLocalPersonalDetails(details);
 }
 
+/**
+ * Fetches the Currency preferences based on location
+ * @param {bool} withLocation
+ */
+function fetchCurrencyPreferences(withLocation) {
+    let coords = {};
+    let currency = '';
+
+    if (withLocation) {
+        Geolocation.getCurrentPosition((position) => {
+            coords = {
+                longitude: position.coords.longitude,
+                latitude: position.coords.latitude,
+            };
+        });
+    }
+
+    API.GetPreferredCurrency({...coords})
+        .then((data) => {
+            currency = data.currency;
+        })
+        .then(API.GetCurrencyList)
+        .then((data) => {
+            const currencyList = JSON.parse(data.currencyList);
+            Onyx.merge(ONYXKEYS.CURRENCY_LIST, currencyList);
+            return currencyList;
+        })
+        .then((currencyList) => {
+            Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS,
```

Comment:
Why are we saving this in `ONYXKEYS.MY_PERSONAL_DETAILS` and not in its own key?

Reply:
@iwiznia This was what was suggested I should do in the original issue.

Reply:
Can you link me to it? I am curious to know...

Reply:
I believe this made sense as we persist timezone to the `MY_PERSONAL_DETAILS` object aready.

Reply:
Yeah, that's another thing I don't quite like, anyway NAB